Source/KSCrash/Source/KSCrash/Recording/BSG_KSCrashReport.c

```diff
@@ -1635,3 +1621,65 @@ void bsg_kscrashreport_logCrash(const BSG_KSCrash_Context *const crashContext) {
     bsg_kscrw_i_logCrashType(crash);
     bsg_kscrw_i_logCrashThreadBacktrace(&crashContext->crash);
 }
+
+static const size_t bsg_g_buffer_increment = sizeof(char) * 8 * 1024;
+static size_t bsg_g_thread_json_size = 128 * 1024;
+static char *bsg_g_thread_json_data;
+
+int bsg_kscrw_i_collectJsonData(const char *const data, const size_t length, void *const userData) {
+    if (bsg_kscrw_i_exceedsBufferLen(length)) {
+        bsg_g_thread_json_size += bsg_g_buffer_increment;
+        bsg_g_thread_json_data = realloc(bsg_g_thread_json_data, bsg_g_thread_json_size);
```

Comment:
It seems to be an anti-pattern to assign the result straight back to the original pointer as it doesn't all you to free the memory. I guess in this case it's a single static area, but would it not be cleaner to `free` and return the error code immediately after `realloc` if it fails rather than introduce a second if-clause to check this later?

Reply:
Implemented suggestion and returned immediately in the error case.